server/src/main/java/org/opensearch/action/resync/ResyncReplicationRequest.java

```diff
@@ -103,7 +103,7 @@ public boolean equals(final Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(trimAboveSeqNo, maxSeenAutoIdTimestampOnPrimary, operations);
+        return Objects.hash(trimAboveSeqNo, maxSeenAutoIdTimestampOnPrimary, Arrays.hashCode(operations));
```

Comment:
We are increasing time complexity by doing this change for sure. Is there a specific reason for doing this?

Reply:
@sandeshkr419 the hash of `operations` here is the array object's hash, but does not hash the contents.  You could have a case of two different arrays containing the same elements, which would produce a different hashCode here but match the `.equals()` check which uses `Arrays.equals()`.  This is a good change.